include/irr/video/IGPUQueue.h

```diff
@@ -0,0 +1,75 @@
+#ifndef __IRR_I_GPU_QUEUE_H_INCLUDED__
+#define __IRR_I_GPU_QUEUE_H_INCLUDED__
+
+#include <irr/core/IReferenceCounted.h>
+#include <irr/video/IGPUPrimaryCommandBuffer.h>
+#include "IDriverFence.h"
+
+namespace irr {
+namespace video
+{
+
+class IGPUQueue : public core::IReferenceCounted
+{
+public:
+    enum E_CREATE_FLAGS : uint32_t
+    {
+        ECF_PROTECTED_BIT = 0x01
+    };
+
+    struct SSubmitInfo
+    {
+        //uint32_t waitSemaphoreCount;
+        //const VkSemaphore* pWaitSemaphores;
+        //const VkPipelineStageFlags* pWaitDstStageMask;
+        //uint32_t signalSemaphoreCount;
+        //const VkSemaphore* pSignalSemaphores;
+        uint32_t commandBufferCount;
+        const IGPUPrimaryCommandBuffer** commandBuffers;
+    };
+
+    //! `flags` takes bits from E_CREATE_FLAGS
+    IGPUQueue(uint32_t _famIx, uint32_t _flags, float _priority)
+        : m_flags(_flags), m_familyIndex(_famIx), m_priority(_priority)
+    {
+
+    }
```

Comment:
same as before, they'll need to be created by a factory so no public constructor necessary